RedRock.Calendar.Modules.Finance.Api/FinanceController.cs

```diff
@@ -14,18 +17,30 @@ namespace RedRock.Calendar.Modules.Finance.Api
     public class FinanceController : ControllerBase
     {
         private readonly IFinanceService financeService;
+        private readonly IConverter converter;
 
-        public FinanceController(IFinanceService financeService)
+        public FinanceController(IFinanceService financeService, IConverter converter)
         {
             this.financeService = financeService;
+            this.converter = converter;
         }
 
+
         [HttpGet]
         public async Task<ActionResult<IEnumerable<FinanceDTO>>> GetMonthlyFee(Guid userReference, DateTime startDate, DateTime endDate)
         {
             var result = await financeService.GetMonthlyFee(userReference, startDate, endDate);
             return (result == null) ? NotFound() : Ok(result);
         }
 
+
+        [HttpGet("{id}/pdf")]
+        public async Task<IActionResult> CreatePDF(Guid id, DateTime startDate, DateTime endDate)
+        {
+            var pdf = await financeService.GetFeeDocument(id, startDate, endDate);
+            var file = converter.Convert(pdf);
```

Comment:
Move this conversion into the financeService.GetFeeDocument, the controller should not contain any logic